pkg/resource/resource.go

```diff
@@ -175,81 +176,109 @@ func CreateFromYAML(yamlFile []byte,
+	obj := &unstructured.Unstructured{
+		Object: map[string]interface{}{},
+	}
 
-		// kernel affinity related attributes only set if there is an
-		// annotation specialresource.openshift.io/kernel-affine: true
-		if kernel.IsObjectAffine(obj) {
-			if err = kernel.SetAffineAttributes(obj, kernelFullVersion, operatingSystemMajorMinor); err != nil {
-				return fmt.Errorf("cannot set kernel affine attributes: %w", err)
-			}
-		}
+	jsonSpec, err := yaml.YAMLToJSON(yamlSpec)
+	if err != nil {
+		return fmt.Errorf("Could not convert yaml file to json: %s: error %w", string(yamlSpec), err)
+	}
 
-		// Add nodeSelector terms for the specialresource
-		// we do not want to spread HW enablement stacks on all nodes
-		if err = SetNodeSelectorTerms(obj, nodeSelector); err != nil {
-			return fmt.Errorf("setting NodeSelectorTerms failed: %w", err)
-		}
+	if err = obj.UnmarshalJSON(jsonSpec); err != nil {
+		return fmt.Errorf("cannot unmarshall json spec, check your manifest: %s: %w", jsonSpec, err)
+	}
 
-		// We are only building a driver-container if we cannot pull the image
-		// We are asuming that vendors provide pre compiled DriverContainers
-		// If err == nil, build a new container, if err != nil skip it
-		if err = rebuildDriverContainer(obj); err != nil {
-			log.Info("Skipping building driver-container", "Name", obj.GetName())
-			return nil
-		}
+	yamlKind := obj.GetKind()
+	metricValue := 0
+	defer func() {
+		metrics.SetCompletedKind(name, yamlKind, metricValue)
+	}()
 
-		// Callbacks before CRUD will update the manifests
-		if err = BeforeCRUD(obj, owner); err != nil {
-			return fmt.Errorf("before CRUD hooks failed: %w", err)
-		}
-		// Create Update Delete Patch resources
-		err = CRUD(obj, releaseInstalled, owner, name, namespace)
-		// The mutating webhook needs a couple of secs to be ready
-		// sleep for 5 secs and requeue
-		if err != nil {
-			if strings.Contains(err.Error(), "failed calling webhook") {
-				return fmt.Errorf("webhook not ready, requeue: %w", err)
-			}
+	//  Do not override the namespace if alreayd set
+	if IsNamespaced(obj.GetKind()) && obj.GetNamespace() == "" {
+		log.Info("Namespace empty settting", "namespace", namespace)
+		obj.SetNamespace(namespace)
+	}
 
-			return fmt.Errorf("CRUD exited non-zero on Object: %+v: %w", obj, err)
+	// We used this for predicate filtering, we're watching a lot of
+	// API Objects we want to ignore all objects that do not have this
+	// label.
+	if err = filter.SetLabel(obj); err != nil {
+		return fmt.Errorf("could not set label: %w", err)
+	}
+	// kernel affinity related attributes only set if there is an
+	// annotation specialresource.openshift.io/kernel-affine: true
+	if kernel.IsObjectAffine(obj) {
+		if err = kernel.SetAffineAttributes(obj, kernelFullVersion, operatingSystemMajorMinor); err != nil {
+			return fmt.Errorf("cannot set kernel affine attributes: %w", err)
 		}
+	}
+
+	// Add nodeSelector terms for the specialresource
```

Comment:
Do I understand correctly that this selector is from the SR?
```suggestion
	// Add nodeSelector terms from the specialresource
```

Reply:
the nodeSelector is defined in the CR spec, but the phrase can be under as add nodeSelector defined for the SR to the object, i will change the description accordingly